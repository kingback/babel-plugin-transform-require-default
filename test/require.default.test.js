const babel = require('babel-core');
const plugin = require('../index');

var example = `
a.require("xxx");
arequire("xxx");
require(ddd);
require('');
require();
require('xxx');
require('xxx')();
require(111);
require('xxx').default;
require('xxx')['default'];
`;

it('require default', () => {
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});