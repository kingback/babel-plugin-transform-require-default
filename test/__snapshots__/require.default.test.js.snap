// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require default 1`] = `
"function __require_default_func__(module) {
  return module && module.__esModule ? module[\\"default\\"] : module;
}

a.require(\\"xxx\\");
arequire(\\"xxx\\");
require(ddd);
require('');
require();
__require_default_func__(require(\\"xxx\\"));
__require_default_func__(require(\\"xxx\\"))();
require(111);
require('xxx').default;
require('xxx')['default'];"
`;
